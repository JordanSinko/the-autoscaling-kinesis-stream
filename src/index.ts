import { Construct } from "@aws-cdk/core";
import { Stream } from '@aws-cdk/aws-kinesis';

export class AutoscalingKinesisStream extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);
    new Stream(this, 'Stream', {});
  }
}
