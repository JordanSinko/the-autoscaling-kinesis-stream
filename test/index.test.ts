import "@aws-cdk/assert/jest";
import { App, Stack } from "@aws-cdk/core";

import { AutoscalingKinesisStream } from "../src";

test("Snapshot", () => {
  const app = new App();
  const stack = new Stack(app, "testing-stack");

  new AutoscalingKinesisStream(stack, "Stream");

  expect(app.synth().getStackArtifact(stack.artifactId).template).toMatchSnapshot();
});
