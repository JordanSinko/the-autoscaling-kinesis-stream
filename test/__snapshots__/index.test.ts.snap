// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Object {
  "Conditions": Object {
    "AwsCdkKinesisEncryptedStreamsUnsupportedRegions": Object {
      "Fn::Or": Array [
        Object {
          "Fn::Equals": Array [
            Object {
              "Ref": "AWS::Region",
            },
            "cn-north-1",
          ],
        },
        Object {
          "Fn::Equals": Array [
            Object {
              "Ref": "AWS::Region",
            },
            "cn-northwest-1",
          ],
        },
      ],
    },
  },
  "Resources": Object {
    "Stream862536A4": Object {
      "Properties": Object {
        "RetentionPeriodHours": 24,
        "ShardCount": 1,
        "StreamEncryption": Object {
          "Fn::If": Array [
            "AwsCdkKinesisEncryptedStreamsUnsupportedRegions",
            Object {
              "Ref": "AWS::NoValue",
            },
            Object {
              "EncryptionType": "KMS",
              "KeyId": "alias/aws/kinesis",
            },
          ],
        },
      },
      "Type": "AWS::Kinesis::Stream",
    },
  },
}
`;
